ShopifyTests/API/Mocks/GraphClientMock.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  GraphClientMock.swift
+//  ShopifyTests
+//
+//  Created by Evgeniy Antonov on 4/10/18.
+//  Copyright © 2018 RubyGarage. All rights reserved.
+//
+
+import ShopApp_Gateway
+
+@testable import MobileBuySDK
+@testable import Shopify
+
+class GraphClientMock: Graph.Client {
+    var returnedResponse: Storefront.QueryRoot?
+    var returnedError: Graph.QueryError?
+    
+    override func queryGraphWith(_ query: Storefront.QueryRootQuery, cachePolicy: Graph.CachePolicy?, retryHandler: Graph.RetryHandler<Storefront.QueryRoot>?, completionHandler: @escaping Graph.QueryCompletion) -> Task {
+        
+        completionHandler(returnedResponse, returnedError)
+        
+        let client = Graph.Client(shopDomain: "shopDomain", apiKey: "apiKey")
+        let queryGraph = Storefront.buildQuery { $0
+            .shop { $0
+                .name()
+            }
+        }
+            
+        let task = client.queryGraphWith(queryGraph) { (response, error) in }
```

Comment:
Can we not create client object and use `super`? Also can we not create `task` variable and return value at once?